extensions/amp-base-carousel/1.0/scroller.js

```diff
@@ -357,6 +363,11 @@ function renderSlides(
   classes
 ) {
   const {length} = children;
+  const lightboxProps = openLightbox && {
```

Comment:
I'm curious if it'd make sense to make `WithLightbox` to wrap each slide inside for lightboxable cases? The `WithLightbox`, it looks like, already implements something like this.

Reply:
Discussed offline, there are a few reasons this approach is preferred to wrapping each slide:
- [Partial rendering inside `BaseCarousel`](https://github.com/ampproject/amphtml/blob/46b225176dd40c05591132e5f61bd4f6ac44769f/extensions/amp-base-carousel/1.0/base-carousel.js#L376): For the portion of slides that are not rendered, they also would not be registered with the `LightboxGallery`, so if it were to open, you would only see images for some of the lightbox consumers, which is not desirable for the grid view.
- [Slide reordering inside `BaseCarousel`](https://github.com/ampproject/amphtml/blob/fac0f5f629752c204388b7f5fadd1d4aad01ac47/extensions/amp-base-carousel/1.0/scroller.js#L389-L408): Looping slides are rendered in a different order than given. This means that if you give slides `A`, `B`, `C` and carousel renders them as `B`, `C`, `A` for looping, the lightbox gallery would then also receive them in `B`, `C`, `A`, order, even though - again for grid view - they should be registered to the `LightboxGallery` in the same preserved order in which they are given to the `BaseCarousel`.

For these reasons, wrapping each slide leaks implementation details.

However, this opens up the next point of investigation: Is it okay to reuse JSX? Because `BaseCarousel` provides `children` to `WithLightbox` both as `children` and `render={() => children}`, we need to understand the full consequences of using them this way. i.e. What happens to `ref`s?